src/containers/shared/components/Transaction/XChainClaim/parser.ts

```diff
@@ -0,0 +1,22 @@
+export function parser(tx: any, meta: any) {
```

Comment:
Could you add a comment explaining the pre-conditions for using this function? (e.g. What type of transaction meta-data is required here?) - I'm having a bit of a hard time guessing how I could use this function correctly.

Reply:
cc @ckniffen - the new transaction format needs a bunch more documentation, but that's out of scope of this ticket. It takes the transaction shape and the metadata shape as displayed in the `tx` output.

Reply:
@JST5000 @ckniffen Does this clarify it better?

Reply:
I think I'm clear on what the parser does, it just seemed like this function wouldn't produce the correct results if you passed in any transaction. And since it required a specific kind of transaction in order to produce the right results, I was thinking that'd be good to include as a comment here. (Otherwise I may accidentally treat it's type signature as the only limitation on it's use)

Reply:
Yeah there are still some refactors that need to be made to the transaction system to really make it easy to read/use. But the current system is so much better than how it used to be.